### ISSUE TITLE
 Help for TTS and Vocoder

### ISSUE THREAD

**Issue by AyseEe501** [CorentinJ/Real-Time-Voice-Cloning#770]
Hello first of all.I conducted a synthesis training from the Turkish dataset I created.
As the Loss value decreases for 205 data, it goes very well. However, when I run demo_toolbox.py over the cpu, I cannot understand the sounds.Is this because I use Encoder and Vocoder with trained data. Should I train them myself to be able to do TTS. I would be very happy if you could give an idea about this. I still have some deficiencies in this regard.

![step-4500-mel-spectrogram_sample_1](https://user-images.githubusercontent.com/81026327/121594764-2b6cbc80-ca46-11eb-900c-e7f62a755f51.png)

![attention_step_4500_sample_1](https://user-images.githubusercontent.com/81026327/121594783-33c4f780-ca46-11eb-8358-e00fb1964601.png)

Also, there should be a data line in the second image but it doesn't show up in this image. I still don't understand where I went wrong.

Apart from all these, when I try to do vocoder training, I get an error message that it cannot find the file in it. While it should work normally, the vocoder should be created in sv2tts, but this file does not create an error. What should I do for this?
![image](https://user-images.githubusercontent.com/81026327/121596364-01b49500-ca48-11eb-9e95-740aa7a4cc2d.png)

**Comment by blue-fish** [comment]
> Also, there should be a data line in the second image but it doesn't show up in this image. I still don't understand where I went wrong.

The absence of the line means attention has not been learned yet. (Attention is needed for inference to work.) In my experience, it takes anywhere between 10000 and 30000 steps to learn attention using the LibriSpeech dataset. Keep training. If you still don't see the line by 50000 steps, verify the quality of your dataset and double-check the model configuration.

> When I try to do vocoder training, I get an error message that it cannot find the file in it. 

It looks like Python doesn't have permission to make the folder called `Veri\SV2TTS\vocoder\mels_gta`. Try making the folder yourself and try the preprocess again.

There is a known bug with vocoder preprocess. You may need to edit a file to make it work: https://github.com/CorentinJ/Real-Time-Voice-Cloning/issues/729#issuecomment-816901953

**Comment by AyseEe501** [comment]
Thank you very much for your help.I'm currently training on my own dataset and it predicts very well.When it finds Attention,like the predicted sounds,Is it understandable readable in the text synthesized in demo_toolbox?

**Comment by blue-fish** [comment]
Yes. After attention is learned, you can use demo_toolbox to synthesize new speech.

**Comment by EkinUstundag** [comment]
> Hello first of all.I conducted a synthesis training from the Turkish dataset I created.
> As the Loss value decreases for 205 data, it goes very well. However, when I run demo_toolbox.py over the cpu, I cannot understand the sounds.Is this because I use Encoder and Vocoder with trained data. Should I train them myself to be able to do TTS. I would be very happy if you could give an idea about this. I still have some deficiencies in this regard.
> 
> ![step-4500-mel-spectrogram_sample_1](https://user-images.githubusercontent.com/81026327/121594764-2b6cbc80-ca46-11eb-900c-e7f62a755f51.png)
> 
> ![attention_step_4500_sample_1](https://user-images.githubusercontent.com/81026327/121594783-33c4f780-ca46-11eb-8358-e00fb1964601.png)
> 
> Also, there should be a data line in the second image but it doesn't show up in this image. I still don't understand where I went wrong.
> 
> Apart from all these, when I try to do vocoder training, I get an error message that it cannot find the file in it. While it should work normally, the vocoder should be created in sv2tts, but this file does not create an error. What should I do for this?
> ![image](https://user-images.githubusercontent.com/81026327/121596364-01b49500-ca48-11eb-9e95-740aa7a4cc2d.png)


Hi, I'm Ekin. I'm from Turkey. I'm interested in doing Voice-Cloning in Turkish. If you are still working on it, can I help ?

**Comment by AyseEe501** [comment]
Hi Ekin.Firstlly nice to meet you.I would really appreciate if you could help me. I'm still having trouble with some points.

> > Hello first of all.I conducted a synthesis training from the Turkish dataset I created.
> > As the Loss value decreases for 205 data, it goes very well. However, when I run demo_toolbox.py over the cpu, I cannot understand the sounds.Is this because I use Encoder and Vocoder with trained data. Should I train them myself to be able to do TTS. I would be very happy if you could give an idea about this. I still have some deficiencies in this regard.
> > ![step-4500-mel-spectrogram_sample_1](https://user-images.githubusercontent.com/81026327/121594764-2b6cbc80-ca46-11eb-900c-e7f62a755f51.png)
> > ![attention_step_4500_sample_1](https://user-images.githubusercontent.com/81026327/121594783-33c4f780-ca46-11eb-8358-e00fb1964601.png)
> > Also, there should be a data line in the second image but it doesn't show up in this image. I still don't understand where I went wrong.
> > Apart from all these, when I try to do vocoder training, I get an error message that it cannot find the file in it. While it should work normally, the vocoder should be created in sv2tts, but this file does not create an error. What should I do for this?
> > ![image](https://user-images.githubusercontent.com/81026327/121596364-01b49500-ca48-11eb-9e95-740aa7a4cc2d.png)
> 
> Hi, I'm Ekin. I'm from Turkey. I'm interested in doing Voice-Cloning in Turkish. If you are still working on it, can I help ?

**Comment by EkinUstundag** [comment]
> Hi Ekin.Firstlly nice to meet you.I would really appreciate if you could help me. I'm still having trouble with some points.

> > Hi, I'm Ekin. I'm from Turkey. I'm interested in doing Voice-Cloning in Turkish. If you are still working on it, can I help ?

I would love to. Where can I get the current state of the project?
Also can I get any kind of a contact info of yours? like email ?

**Comment by AyseEe501** [comment]
Firstlly this is  my e-mail ->aysenurcakmakci53@hotmail.com
First of all, I increased the dataset from 205 to 666. Because I can't find the attention layer.That's already the problem. Learning rate increased from 0.001 to 0.005 for the second stage.But still can't find.Do I need to clean the dataset or how long should I wait I dont know.
![image](https://user-images.githubusercontent.com/81026327/132002156-0fb39f69-515c-44f4-9bbd-aa9bdc3ab71b.png)

For 26500k step:
![image](https://user-images.githubusercontent.com/81026327/132002737-873819c8-d862-4fd2-9022-11730d76145c.png)

However, the loss value is low and good at predicting sound
![image](https://user-images.githubusercontent.com/81026327/132002814-79059c1c-ccb4-4a47-b78d-ff5ad630a09d.png)




> > Merhaba Ekin.Öncelikle tanıştığıma memnun oldum.Yardımcı olabilirseniz çok sevinirim. Bazı noktalarda hala sorun yaşıyorum.
> 
> > > Merhaba ben Ekin. Ben Türkiyeliyim. Türkçe Ses Klonlama yapmakla ilgileniyorum. Hala çalışıyorsanız yardımcı olabilir miyim?
> 
> İsterdim. Projenin mevcut durumunu nereden öğrenebilirim?
> Ayrıca her türlü iletişim bilgilerinizi alabilir miyim? e-posta gibi mi?
